lib/rules/prefer-destructuring.js

```diff
@@ -15,19 +15,59 @@ module.exports = {
             category: "ECMAScript 6",
             recommended: false
         },
-
         schema: [
             {
-                type: "object",
-                properties: {
-                    array: {
-                        type: "boolean"
+
+                // old support {array: Boolean, object: Boolean}
+                // new support {VariableDeclarator: {}, AssignmentExpression: {}}
+                oneOf: [
+                    {
+                        type: "object",
+                        properties: {
+                            VariableDeclarator: {
+                                type: "object",
+                                properties: {
+                                    array: {
+                                        type: "boolean",
+                                        default: true
```

Comment:
(Continuing the discussion from https://github.com/eslint/eslint/pull/8796#discussion_r125576512)

If it has no effects, then I think we should remove it, because it could be misleading and make people think that it does have an effect. If we want to indicate the default there, I think adding a comment would be better.